src/commands/deploy/setPreDeployTaskForDotnet.ts

```diff
@@ -0,0 +1,155 @@
+    const preDeployTaskSetting: string = 'preDeployTask';
+    const showPreDeployWarningSetting: string = 'showPreDeployWarning';
+    const workspaceFspath: string = context.workspace.uri.fsPath;
+
+    // don't overwrite preDeploy or deploySubpath if it exists and respect configurePreDeployTasks setting
+    if (!getWorkspaceSetting<boolean>(showPreDeployWarningSetting, workspaceFspath)
+        || getWorkspaceSetting<string>(preDeployTaskSetting, workspaceFspath)
+        || getWorkspaceSetting<string>(constants.configurationSettings.deploySubpath, workspaceFspath)) {
+        return;
+    }
+
+    // if the user is deploying a different folder than the root, use this folder without setting up defaults
+    if (!isPathEqual(context.deployFsPath, workspaceFspath)) {
+        return;
+    }
+
+    const csprojFile: string | undefined = await tryGetCsprojFile(context, workspaceFspath);
+
+    // if we found a .csproj file set the tasks and workspace settings
+    if (csprojFile) {
+        const targetFramework: string | undefined = await tryGetTargetFramework(csprojFile);
+        context.telemetry.properties.tfw = targetFramework ? targetFramework : 'N/A';
+        if (!targetFramework) {
+            // if the target framework cannot be found, don't try to set defaults
+            return;
+        }
+
+        const notConfiguredForDeploy: string = `Required assets to build and deploy are missing from "${context.workspace.name}".`;
+        const addAssetsButton: MessageItem = { title: "Add Assets" };
+        const dontShowAgainButton: MessageItem = { title: "Don't Show Again" };
+        const input: MessageItem = await ext.ui.showWarningMessage(notConfiguredForDeploy, { modal: true }, addAssetsButton, dontShowAgainButton);
+        if (input === dontShowAgainButton) {
+            await updateWorkspaceSetting(showPreDeployWarningSetting, false, workspaceFspath);
+        } else {
+            // resolves to "."if it is not a subfolder
+            const subfolder: string = path.dirname(path.relative(workspaceFspath, csprojFile));
+
+            // always use posix for debug config because it's committed to source control and works on all OS's
+            const deploySubpath: string = path.posix.join(subfolder, 'bin', 'Debug', targetFramework, 'publish');
+
+            await updateWorkspaceSetting(preDeployTaskSetting, 'publish', workspaceFspath);
+            await updateWorkspaceSetting(constants.configurationSettings.deploySubpath, deploySubpath, workspaceFspath);
+
+            // update the deployContext.deployFsPath with the .NET output path since getDeployFsPath is called prior to this
+            context.deployFsPath = path.join(workspaceFspath, deploySubpath);
+
+            const existingTasks: tasks.ITask[] = tasks.getTasks(context.workspace);
+            const publishTask: tasks.ITask | undefined = existingTasks.find(t1 => {
+                return t1.label === 'publish';
```

Comment:
make a const for the publish task id to be used here and when creating the task below. Same for clean